eng/scripts/Create-APIView.ps1

```diff
@@ -29,17 +32,61 @@ function Get-SwaggerReadMeFile {
       }
 
       $readmeFile = Get-ChildItem -Path $currentPath -Filter "readme.md" -File -ErrorAction SilentlyContinue
-      if ($readmeFile -and $readmeFile.Name -eq "readme.md") {
+      if ($readmeFile -and $readmeFile.Name -eq "readme.md" -and (Assert-AutoRestConfiguration -readMeFile $readmeFile.FullName)) {
           return $readmeFile.FullName
       }
     }
 
     return $null
 }
 
+<#
+.DESCRIPTION
+  Verify if a readMefile is useful autorest configuration
+
+.PARAMETER ReadmeFile
+  Path to the readme file.
+.OUTPUTS
+  true if valid config false otherwise
+#>
+function Assert-AutoRestConfiguration {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$readMeFile
+    )
+
+    $headingsMatch = 0
+    $tagMatch = 0
+    $markDownContent = Get-Content -Path $readMeFile
```

Comment:
We should try and avoid opening the readme file more than once. 

I wrote a basic readme parser at https://github.com/Azure/azure-sdk/blob/main/eng/scripts/Update-Spec-Versions.ps1#L275 for our spec indexing process. We should consider using a shared version but for now we should at least try to avoid reading the files multiple times.

Reply:
Am not sure it's possible to avoid this since we need to probe the readme on source branch then on target branch.